webapp/channels/src/components/channel_threads_rhs/channel_threads_rhs.tsx

```diff
@@ -0,0 +1,266 @@
+            });
+        }
+        if (selected === Tabs.USER) {
+            return formatMessage({
+                id: 'channel_threads.noResults.created',
+                defaultMessage: 'You don’t have any threads that you created in this channel.',
+            });
+        }
+        return formatMessage({
+            id: 'channel_threads.noResults.all',
+            defaultMessage: 'There are no threads in this channel.',
+        });
+    }, [selected]);
+
+    return (
+        <div
+            id='rhsContainer'
+            className='sidebar-right__body ChannelThreadList'
+        >
+            <Header
+                canGoBack={canGoBack}
+                channel={channel}
+                goBack={actions.goBack}
+                isExpanded={isSideBarExpanded}
+                onClose={actions.closeRightHandSide}
+                toggleRhsExpanded={actions.toggleRhsExpanded}
+            />
+            <div className='tab-buttons'>
+                <div className='tab-button-wrapper'>
+                    <Button
+                        className='Button___large Margined'
+                        isActive={selected === Tabs.ALL}
+                        onClick={makeHandleTab(Tabs.ALL)}
+                    >
+                        <FormattedMessage
+                            id='channel_threads.filters.all'
+                            defaultMessage='All'
+                        />
+                    </Button>
+                </div>
+                <div className='tab-button-wrapper'>
+                    <Button
+                        className='Button___large Margined'
+                        isActive={selected === Tabs.FOLLOWING}
+                        onClick={makeHandleTab(Tabs.FOLLOWING)}
+                    >
+                        <FormattedMessage
+                            id='channel_threads.filters.following'
+                            defaultMessage='Following'
+                        />
+                    </Button>
+                </div>
+                <div className='tab-button-wrapper'>
+                    <Button
+                        className='Button___large Margined'
+                        isActive={selected === Tabs.USER}
+                        onClick={makeHandleTab(Tabs.USER)}
+                    >
+                        <FormattedMessage
+                            id='channel_threads.filters.createdByMe'
+                            defaultMessage='Created by me'
+                        />
+                    </Button>
+                </div>
+            </div>
+            <div className='channel-threads'>
+                {(isLoading && ids.length === 0) && (
+                    <LoadingScreen style={loadingStyle}/>
+                )}
+
+                {ids.length ? (
+                    <VirtualizedThreadList
+                        key={`${selected}-${Math.min(totalThreads, Constants.THREADS_PAGE_SIZE)}`}
```

Comment:
A bit of background:
When we receive a new thread we are checking if the thread is newer than at least one of the existing threads, and if it isn't then we add it, this is done to not create gaps in pagination. The older items will get fetched by scrolling either way.

Now on the actual issue. 
If you have only one page of items advancing the total doesn't make the `InfiniteLoader` to request for the next page.

One solution would be to change the `shouldAddThread` to take into account the `total` value of threads, and add it if it has one page without checking seniority. This way we won't need to fetch the next page and we just show the newly added thread.

The other solution would be using a `key` to un-mount and re-mount the component, which will make the `InfiniteLoader` request for the next page.

Reply:
@hmhealey well the issue is replicating when you are already in a tab open, 
and you have less than one page of threads in total. 
When a new thread is fetched via whatever means
we won't add it if it's older than all the other threads in the list 
because we are relying on the InfiniteLoader to add it when we scroll.
But in this particular case we can't scroll since we have less than one page, 
and the loader won't trigger an request if the total changed it seems.
BTW, if we switch tabs it will be called.